activestorage/lib/active_storage/service/gcs_service.rb

```diff
@@ -160,7 +176,39 @@ def bucket
       end
 
       def client
-        @client ||= Google::Cloud::Storage.new(**config.except(:bucket, :cache_control))
+        @client ||= Google::Cloud::Storage.new(**config.except(:bucket, :cache_control, :iam, :gsa_email))
+      end
+
+      def issuer
+        @issuer ||= begin
+          if @config[:gsa_email]
+            @config[:gsa_email]
+          else
+            uri = URI.parse("http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/email")
+            http = Net::HTTP.new(uri.host, uri.port)
+            request = Net::HTTP::Get.new(uri.request_uri)
+            request["Metadata-Flavor"] = "Google"
+
+            http.request(request).body
+          end
+        end
```

Comment:
```suggestion
        @issuer ||= if @config[:gsa_email]
                      @config[:gsa_email]
                    else
                      uri = URI.parse("http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/email")
                      http = Net::HTTP.new(uri.host, uri.port)
                      request = Net::HTTP::Get.new(uri.request_uri)
                      request["Metadata-Flavor"] = "Google"

                      http.request(request).body
                    end
```
begin is necessary in this case?